sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarPartitionReaderWithPartitionValues.scala

```diff
@@ -95,6 +110,55 @@ object ColumnarPartitionReaderWithPartitionValues {
     result
   }
 
+  /**
+   * Splits each Gpu Column Vector into multiple batches based on row counts.
+   *
+   * Example,
+   * columnVectors = [ Cv('pqr', 55), Cv('111', 55), Cv('ab1', 55)]
+   * rowCounts = [ 20, 10, 15, 10 ]
+   * splitIndices = [ 20, 30, 35 ]
+   * Result:
+   *    [
+   *      [Cv('pqr',20), Cv('pqr',10), Cv('pqr',15), Cv('pqr',10)],
+   *      [Cv('111',20), Cv('111',10), Cv('111',15), Cv('111',10)],
+   *      [Cv('ab1',20), Cv('ab1',10), Cv('ab1',15), Cv('ab1',10)]
+   *    ]
+   */
+  private def splitColumnVector(columnVectors: Seq[ColumnVector],
+      rowCounts: Array[Long]): Seq[Array[GpuColumnVector]] = {
```

Comment:
Rather than using Array[GpuColumnVector] everywhere, wondering if the code would be easier to read/follow if we used Seq[ColumnarBatch] instead.  This will also make it easier to have the data be spillable for the portions of a split that are not immediately used.  Seems like this should really be splitBatch not splitColumnVector(s).

Reply:
That is true. I will refactor the intermediate methods to use `ColumnarBatch` instead

For the method name, since it is splitting a sequence of ColumnVector, I have renamed it to `splitColumnVectorsToBatches()`

Reply:
It should just be written as splitBatch, taking a single batch in and returning an Array[ColumnarBatch].  The calling code already has the batch and goes out of its way to build Array[GpuColumnVector], so let's just skip that step and pass the batch directly.